### PR TITLE
:bug: Fix --force option to recreate the files by kubebuilder create api

### PR DESCRIPTION
**Pull request by prafull01**
**Description**
Added changes to overwrite the files generated by kubebuilder create api when --force option is provided.

**Motivation**
--force command is failing on kubebuilder
Fixes: #1845

### CODE REVIEW

File: pkg/plugins/golang/v2/api.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (p *createAPISubcommand) GetScaffolder() (cmdutil.Scaffolder, error) {
 
 	// Create the actual resource from the resource options
 	res := p.resource.NewResource(p.config, p.doResource)
-	return scaffolds.NewAPIScaffolder(p.config, string(bp), res, p.doResource, p.doController, plugins), nil
+	return scaffolds.NewAPIScaffolder(p.config, string(bp), res, p.doResource, p.doController, p.force, plugins), nil
```
Comment (camilamacedo86) [inline pkg/plugins/golang/v2/api.go:180]: Do you not need the same fix for webhooks as well?

**Review by camilamacedo86 — APPROVED** [review]
it shows fine for me. So, 

/approve

Do you not need the same fix for webhooks as well? Did you check their behaviour too?

**Comment by prafull01** [comment]
@camilamacedo86 There is no --force option in create webhook command, hence for webhook this will be Feature request. Hence, that should be done in a separate PR and only for V3.

File: testdata/project-v3-config/main.go [hunk 2]
```diff
@@ -77,6 +77,14 @@ func main() {
...
 	}
 
+	if err = (&controllers.CaptainReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("Captain"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "Captain")
+		os.Exit(1)
+	}
```
Comment (camilamacedo86) [inline testdata/project-v3-config/main.go:87]: /hold

I should not duplicate it since the --force will re-create the file.

Comment (camilamacedo86) [reply testdata/project-v3-config/main.go:87]: Thread: https://github.com/kubernetes-sigs/kubebuilder/pull/1903#discussion_r542919409

/hold cancel

**Comment by camilamacedo86** [comment]
we bug triage the issue today. So, it shows fine we 

/lgtm 

this one. Very reasonable for me as well.

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Pull request merged**